canvasapi/canvas_object.py

```diff
@@ -60,6 +62,16 @@ def set_attributes(self, attributes):
         """
         for attribute, value in attributes.items():
             self.__setattr__(attribute, value)
+            if attribute == "content-type":
+                self.__setattr__("content_type", value)
+                warnings.warn(
+                    (
+                        "The 'content-type' attribute will be removed "
+                        "in a future version. Please use "
+                        "'content_type' instead."
+                    ),
+                    UserWarning,
+                )
```

Comment:
Upon re-reading the discussion, was the intention to warn users who were trying to access `content-type` via `getattr`? In that case, a warning could make sense but it would have to be on the getter function, not the setter function as it is here.

I think modifying `__getattribute__` could do the trick?

Reply:
Thanks @Thetwam I just looekd at this too and was thinking the same, that it's probably better to have this in a `__getattribute__` or possibly even better in `__getattr__ ` rather than in the `set_attributes`. Then you don't lose a similarly named attribute with. 

`__getattr__` would only catch the ones that don't exist so wouldn't need to run through for everything. 

I think it's unlikely they'd have a content_type and content-type but it's possible for other headers.

Reply:
TIL there's difference between `__getattribute__` and `__getattr__`.

However, if I'm understanding it correctly, we'd still want to use `__getattribute__` for the time being, since the current behavior is to keep `content-type` while adding `content_type`.

I moved the warning to `__getattribute__`, which passes the `test__getattribute__content_type_warns` test, whereas `__getattr__` fails (doesn't throw the expected warning)

Reply:
Right `__getatribute__` is called anytime an attribute is accessed
`__getattr__` is  called when the item is not found via `__getattribute__`

I was thinking of something more like this change why does this have to be specific for content-type? And why the deprecation warning. This idea is that if there's a value called "content-type" it will return it, otherwise __getattr__ is called and it will try to return content-type instead of content_type. If not it returns nothing. This means the value with the dashes always works, if there's a value with dashes and underscore you HAVE to get use getattr directly. But it's still here for the convenience. That seems like it was the initial request. 

https://github.com/ucfopen/canvasapi/commit/8eae2d5a9940f339e46cbaa4960438078e982c82